### DISCUSSION TITLE
application or network url won't open in the browser

### DISCUSSION THREAD

**Discussion by johnquest** [flyntwp/flynt#526] [Q&A] [answered]
I'm running WP locally in a LEMP stack. When I do npm start, both the application url at http://mysite.local:3000 and the network address at http://localhost:5173 won't open in the browser. Do I need to tweak nginx so it serves these addresses and do I need firewall rules to open port 3000?

**Comment by timohubois** [comment]
@johnquest technically the application url is the value of the const `wordpressHost` inside the `vite.conf.js` file. 

It's value should be set to the URL of your local site (with or without port, depending on your setup).

**Comment by johnquest** [comment]
I see that. npm run build works but not npm start. In the wordpressHost address, the links to the scripts and stylesheets change to be http://localhost:5173/assets/main.js for example, but this address isn't available outside of the docker container where it runs after running npm start. What I'm wondering is for those that develop with flynt with docker, what is the configuration of the containers to get this address exposed so the scss files work in the wordpressHost address?

**Comment by johnquest** [comment]
I've figured this out so that vite works with ddev. In vite.config "host" needs to be set to the ip address of the web docker container, not "localhost." An origin address or port needs to be added -> http://mysite.ddev.site:5173 with the wordpressHost set to http://mysite.ddev.site.

**Comment by timohubois** [comment, marked as answer]
@johnquest glad to read, that you got it working with DDEV! 

For pure docker the solution should be similar and an extra port (in your case 5173 the default vite server port) needs to be exposed for hmr.

Can you maybe write an update to your open issue https://github.com/flyntwp/flynt/issues/525 and close it if your problem is solved?
